### PR TITLE
tests: Re-add per-platform image tests

### PR DESCRIPTION
**Pull request by Aaron1011**
Reverts #7267

The image tests for the upcoming 'DisplayObject.stageRect' support
differ between Linux and Windows, so we need this support again.

To avoid the Linux filename churn that we previously encountered,
we now only include the platform and graphics backend in the filename
(e.g. `expected-linux-Vulkan`). This may result in some unexpected
'mismatched image' test failures if GHA updates to a version of Lavapipe
that changes rendering output, but this should be relatively easy to
notice.

### CODE REVIEW

File: .github/workflows/test_rust.yml [hunk 1]
```diff
@@ -99,7 +99,7 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: swf_images
-          path: tests*/**/actual.png
+          path: tests*/**/actual*.png
```
Comment (relrelb) [inline .github/workflows/test_rust.yml:102]: BTW Why `tests*/` and not `tests/`?

**Comment by relrelb** [comment]
Do you have any idea why `DisplayObject.stageRect` output differs between Windows and Linux? Is it caused by an internal detail of `wgpu`, or maybe we can avoid the difference?

File: .github/workflows/test_rust.yml (hunk 1 above)
Comment (Aaron1011) [reply .github/workflows/test_rust.yml:102]: This is a workaround to preserve the directory structure in the uploaded archive: https://github.com/actions/upload-artifact/issues/206#issuecomment-1157428735

**Comment by Aaron1011** [comment]
@relrelb I haven't investigated the difference yet. However, the `loader_events` test image output differs for me between Vulkan drivers (native NVidia vs lavapipe), so it seems hopeless to expect that all of our image tests should be identical between two different rendering backends (Dx12 vs Vulkan).

**Comment by Herschel** [comment]
To clarify, this is about `scrollRect`, yes? (`stageRect` does not exist)

**Comment by Aaron1011** [comment]
@Herschel My bad - I did mean `scrollRect`

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**